src/pystack/__main__.py

```diff
@@ -319,6 +321,19 @@ def process_core(parser: argparse.ArgumentParser, args: argparse.Namespace) -> N
     if not corefile.exists():
         parser.error(f"Core {corefile} does not exist")
 
+    temp_file = None
+    if corefile.suffix == ".gz":
```

Comment:
Maybe we can improve this to not rely on the filename by checking the gz magic number: `1f 8b` from the header

Reply:
In making this change I hit a small snag: other tests mock the `pathlib.Path().exists()` check, causing them to fail when attempting to open the non-existent corefile in order to check the header. What do you think the correct approach would be here?
- Update those tests to expect the read error?
- Create a corefile fixture?
- Something else?

Reply:
Move this code to a function and mock/patch it in those tests to return None or whatever indicates that this operation doesn't make sense

Reply:
Good call, I should've done this to begin with. Thanks!